### PR TITLE
Pre-processing: Fix for int overflow.

### PR DESCRIPTION
**Pull request by anna-khakimova**
### Details:
 - *Fix for int overflow*

### CODE REVIEW

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp [hunk 1]
```diff
@@ -117,7 +117,12 @@ template<typename DST, typename SRC> static inline DST checked_cast(SRC x) {
 }
 
 static inline Q8_8 mulas(Q0_16 a, U8   s) { return static_cast<Q8_8>((a * s) >>  8); }
-static inline Q8_8 mulaw(Q0_16 a, Q8_8 w) { return static_cast<Q8_8>((a * w) >> 16); }
+static inline Q8_8 mulaw(Q0_16 a, Q8_8 w) {
```
Comment (TolyaTalamanov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp:120]: Can it be one-liner as well ? 
```
return static_cast<Q8_8>((static_cast<uint32_t>(a) * w) >> 16)
```

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp [hunk 2]
```diff
@@ -117,7 +117,12 @@ template<typename DST, typename SRC> static inline DST checked_cast(SRC x) {
...
+   uint32_t mul = a;
+   mul *= w;
+   uint16_t res = static_cast<Q8_8>(mul >> 16);
```
Comment (TolyaTalamanov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp:123]: Also didn't get how to `uint16_t` match to `Q8_8` ? 
What does actually `Q8_8` means ?

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp (hunk 1 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp]: Ok. Done.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp (hunk 2 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp]: `#define Q8_8 uint16_t`

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp (hunk 1 above)
Comment (TolyaTalamanov) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp]: Does it work ?

**Comment by TolyaTalamanov** [comment]
Thanks! LGTM 👍

**Review by TolyaTalamanov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp (hunk 1 above)
Comment (anna-khakimova) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels_impl.hpp]: Yes, it works.

**Pull request merged**